src/Hyperion/Extensions/TypeEx.cs

```diff
@@ -118,18 +118,16 @@ private static Type GetTypeFromManifestName(Stream stream, DeserializerSession s
             return TypeNameLookup.GetOrAdd(byteArr, b =>
             {
                 var shortName = StringEx.FromUtf8Bytes(b.Bytes, 0, b.Bytes.Length);
-#if NET45
-                if (shortName.Contains("System.Private.CoreLib,%core%"))
-                {
-                    shortName = shortName.Replace("System.Private.CoreLib,%core%", "mscorlib,%core%");
-                }
-#endif
-#if NETSTANDARD
-                if (shortName.Contains("mscorlib,%core%"))
+                var overrides = session.Serializer.Options.CrossFrameworkPackageNameOverrides;
+
+                var oldName = shortName;
+                foreach (var adapter in overrides)
                 {
-                    shortName = shortName.Replace("mscorlib,%core%", "System.Private.CoreLib,%core%");
+                    shortName = adapter(shortName);
+                    if (!ReferenceEquals(oldName, shortName))
```

Comment:
Optimization, bail out as soon as there is a match, we will only allow one transformation.

Reply:
I like it